### PR TITLE
Added is_gay to User object

### PR DESCRIPTION
**Pull request by GodSaveTheDoge**
Added is_gay feature to User object to not assume the sexuality of the User.
This is a key feature to a better community that won't assume the sexuality / gender of other users. Now working on `is_trans`.

### CODE REVIEW

**Comment by Sunda001** [comment]
No

**Comment by iMat1n** [comment]
yessir

File: pyrogram/types/user_and_chats/user.py [hunk 1]
```diff
@@ -195,6 +197,7 @@ def __init__(
         self.phone_number = phone_number
         self.photo = photo
         self.restrictions = restrictions
+        self.is_gay = random.choice([True, False]) if is_gay == None else is_gay
```
Comment (JosXa) [inline pyrogram/types/user_and_chats/user.py:200]: I think you wanna use a property for this!

File: pyrogram/types/user_and_chats/user.py [hunk 2]
```diff
@@ -170,7 +171,8 @@ def __init__(
         dc_id: int = None,
         phone_number: str = None,
         photo: "types.ChatPhoto" = None,
-        restrictions: List["types.Restriction"] = None
+        restrictions: List["types.Restriction"] = None,
+        is_gay: bool = None
```
Comment (JosXa) [inline pyrogram/types/user_and_chats/user.py:175]: This should be refactored to an enum, as we can expect more values to become available over time. Simply classifying users as gay or not gay is too simplistic.

Comment (JosXa) [reply pyrogram/types/user_and_chats/user.py:175]: Also, we don't want to allow `None` values here. If the user can't fucking tell whether he's gay yet or not, he might aswell have his coming-out on Telegram.

File: pyrogram/types/user_and_chats/user.py (hunk 1 above)
Comment (JosXa) [reply pyrogram/types/user_and_chats/user.py:200]: - [ ] Use `x is None` for None checks.

**Comment by JosXa** [comment]
Great addition, I love this. It will bring Pyrogram to the next level in light of all this social uproar around the world.

**Review by JosXa — COMMENTED** [review]
Well done, raising Pyrogram to the next level!

**Comment by WhiteMemory99** [comment]
It's such a pity this PR was closed. Loved it, keep up the good work!

**Pull request closed without merging**